docs: importing talos clusters (#72)
siderolabs/docs: public/omni/cluster-management/importing-talos-clusters.mdx

```
--nodes strings                       endpoints of all nodes to import
--talos-version string                talos version of the cluster, if not set, will be detected from the nodes
--kubernetes-version string           kubernetes version of the cluster, if not set, will be detected from the nodes
--initial-talos-version string        initial talos version used on cluster creation, if not set current talos version will be used
--initial-kubernetes-version string   initial kubernetes version used on cluster creation, if not set current kubernetes version will be used
--talosconfig string                  The path to the Talos configuration file. Defaults to 'TALOSCONFIG' env variable if set, otherwise '$HOME/.talos/config'.
--talos-context string                the context to be used for accessing talos. defaults to the selected context in the Talos configuration file
--talos-endpoints strings             override default endpoints in Talos configuration file
```

> To check what will happen during the import process you can run `omnictl cluster import` with `--dry-run` flag:

Reviewer: ```suggestion
To check what will happen during the import process you can run `omnictl cluster import` with the `--dry-run` flag:
```

```

This command does the following:
1. Discover the talos cluster state using provided/discovered information and build an import context to be used during the import.
2. Validate the cluster state to ensure it is healthy and ready to be imported.
3. Save a backup of the current machine configurations to a zip file.
4. Connect the Talos nodes to Omni by applying necessary configurations to the nodes.
5. Create the necessary resources in Omni to represent the imported cluster.

Note that the cluster has been imported but marked as `locked`. This is to prevent any changes being done by Omni to the imported cluster.

### Image Schematic
With [Talos Linux Image Factory](https://factory.talos.dev/), it's possible to create a talos installer using extensions, extra kernel command line arguments as well as other settings. Such configuration is referred to as a Schematic.
Cluster import command tries to detect if a Talos machine was configured using a boot asset from an Image Factory, hence connected to a Schematic. If so, it will try to ensure that Schematic with Omni's connected Image Factory.

> If a Talos machine is not configured using an Image Factory, then the validation step will fail by reporting that the Schematic can't be found.

Reviewer: ```suggestion
If a Talos machine was not configured using an Image Factory asset the validation step will fail by reporting that the Schematic can't be found.
```

Note that if `--force` option was used, Omni will not be able to ensure that it can successfully offer upgrades for these Talos machines while keeping their existing configuration.
Omni will still be able to offer upgrades for these machines, but it will create and use a new Schematic to do so.
If a custom Linux kernel or custom extensions were used while configuring the machine, they will be lost upon upgrade.

This will not affect machines configured by using a boot asset from [Talos releases](https://github.com/siderolabs/talos/releases).

### Config Patches

While importing a cluster to Omni, the idea is to leave the current state of the cluster as is and not to change any configuration on the nodes, unless it's absolutely necessary.
Cluster Machine config patches are used to achieve this idea.

The import command uses initial talos and kubernetes versions to build the default machine config for each node in the cluster.
These configs are then compared to the current machine config on each node to build a config patch.
Each config patch includes the changes made to the machine config since the Talos cluster was created.

> Certain machine config fields are not allowed to be set on Omni managed clusters. These fields are not included in the final config patch.

Author: We do not have these documented, we just have a slice of strings for forbidden fields. Let me see if I can reference it correctly.

After a cluster is imported to Omni, the final machine config for each node is built by applying the config patch to the default machine config built using the initial talos and kubernetes versions used on cluster import.
Redacted version of this machine config can be found under the `Config` tab of the Cluster Machine overview page in the Omni UI, while config patches can be found under `Patches` tab of the same overview page.

### Unlocking the Cluster

Since Omni uses its own Kubernetes apiserver endpoint, when the cluster is unlocked, Omni will update `kube-apiserver`, `kube-scheduler` and `kube-controller-manager` static pod definitions to point to the Omni-managed kubernetes apiserver endpoint.

> ### Aborting an ongoing Import

Reviewer: I think the users will find it useful if you describe the potential risks involved with importing a cluster. A question that pops into my mind is "can I just try importing and abort if it goes wrong, or there might be consequences?"

Author: Since the backup includes all the machine configs, assumption was that anything Omni does is reversible just by applying backed up machine configs.
